### ISSUE TITLE
TypeError: Cannot read property 'httpHost' of undefined

### ISSUE THREAD

**Issue by SteeeveAT** [bwp91/homebridge-ewelink#135]
DUPLICATE TO: https://github.com/bwp91/homebridge-ewelink/issues/133 (sorry)

Hi! got the error "TypeError: Cannot read property 'httpHost' of undefined" and have no idea why this suddenly appears.

Did an update and also NPM is on the latest version
`
[04/11/2020, 09:33:31] [eWeLink] *** Disabling plugin [v3.10.0] ***
[04/11/2020, 09:33:31] [eWeLink] TypeError: Cannot read property 'httpHost' of undefined
    at new eWeLinkWS (/usr/local/lib/node_modules/homebridge-ewelink/lib/ewelink-ws.js:15:25)
    at eWeLinkPlatform.eWeLinkSetup (/usr/local/lib/node_modules/homebridge-ewelink/lib/ewelink-platform.js:79:23)
    at processTicksAndRejections (internal/process/task_queues.js:93:5)
    root : TTY=pts/0 ; PWD=/var/homebridge ; USER=root ; COMMAND=/bin/journalctl -o cat -n 500 -f -u homebridge
pam_unix(sudo:session): session opened for user root by (uid=0)
    root : TTY=unknown ; PWD=/ ; USER=root ; COMMAND=/bin/kill -9 8074
pam_unix(sudo:session): session opened for user root by (uid=0)
pam_unix(sudo:session): session closed for user root
    root : TTY=pts/0 ; PWD=/var/homebridge ; USER=root ; COMMAND=/bin/journalctl -o cat -n 500 -f -u homebridge
pam_unix(sudo:session): session opened for user root by (uid=0)
pam_unix(sudo:session): session closed for user root
    root : TTY=unknown ; PWD=/ ; USER=root ; COMMAND=/bin/kill -9 8210
pam_unix(sudo:session): session opened for user root by (uid=0)
pam_unix(sudo:session): session closed for user root
    root : TTY=pts/0 ; PWD=/var/homebridge ; USER=root ; COMMAND=/bin/journalctl -o cat -n 500 -f -u homebridge
pam_unix(sudo:session): session opened for user root by (uid=0)`

Configuration:
{
            "name": "eWeLink",
            "countryCode": "43",
            "username": "+XXXXX",
            "password": "XXXXX",
            "debug": true,
            "debugReqRes": false,
            "disableHTTPRefresh": false,
            "disablePlugin": false,
            "inUsePowerThreshold": 0,
            "hideTHSwitch": false,
            "lowBattThreshold": 25,
            "sensorTimeDifference": 120,
            "hideZBLDPress": false,
            "resetRFBridge": false,
            "groups": [
                {
                    "operationTime": 100
                }
            ],
            "bridgeSensors": [
                {
                    "sensorTimeLength": 60
                }
            ],
            "platform": "eWeLink"
        }

Any ideas?

Thanks in advance
Stephan

**Comment by bwp91** [comment]
Hi @SteeeveAT 

This seems to be a strange issue affecting a very small number of users. And I'm yet to really find out what's causing it.

But, please can I refer you to [this issue](https://github.com/bwp91/homebridge-ewelink/issues/133) with regards to updating to a beta v3.11.0-4, to see if this fixes it for you.

Also please note the last comment on that issue if you use any light switch devices (eg TX devices)

**Comment by SteeeveAT** [comment]
Hey @bwp91 

Thanks. Did an update to the beta version, no errors, but devices are not responding to on or off :/ 
The state in the home app is updatet.

It looks something like this in the protocol:

```
[04/11/2020, 10:13:44] [eWeLink] Sending WS login request. This text is yellow for clarity.
{
  "action": "userOnline",
  "apikey": "**hidden**",
  "appid": "xxxx",
  "at": "**hidden**",
  "nonce": "eo90ev6q",
  "sequence": "1604481224550",
  "ts": 1604481224,
  "userAgent": "app",
  "version": 8
}
[04/11/2020, 10:13:46] [eWeLink] [11] eWeLink devices loaded from the Homebridge cache.
[04/11/2020, 10:13:46] [eWeLink] [10] primary devices loaded from your eWeLink account.
[04/11/2020, 10:13:46] [eWeLink]  → [Gang] was removed from Homebridge.
[04/11/2020, 10:13:46] [eWeLink]  → [3D Drucker] initialised and found locally with IP [10.0.0.166].
[04/11/2020, 10:13:46] [eWeLink]  → [Sonoff 2 - Snappy] initialised but LAN mode unavailable as device is unreachable.
[04/11/2020, 10:13:46] [eWeLink]  → [Fenstersteckdosen] initialised and found locally with IP [10.0.0.66].
[04/11/2020, 10:13:46] [eWeLink]  → [Sonoff-1] initialised and found locally with IP [10.0.0.71].
[04/11/2020, 10:13:46] [eWeLink]  → [Zirkulationspumpe] initialised and found locally with IP [10.0.0.53].
[04/11/2020, 10:13:46] [eWeLink]  → [Küche] initialised and found locally with IP [10.0.0.185].
[04/11/2020, 10:13:46] [eWeLink]  → [Reserve] initialised but LAN mode unavailable as device is unreachable.
[04/11/2020, 10:13:46] [eWeLink]  → [Gang] initialised and found locally with IP [10.0.0.52].
[04/11/2020, 10:13:46] [eWeLink]  → [Spots Leinwand] initialised and found locally with IP [10.0.0.128].
[04/11/2020, 10:13:46] [eWeLink]  → [Alibert] initialised and found locally with IP [10.0.0.84].
[04/11/2020, 10:13:46] [eWeLink] eWeLink sync complete. Don't forget to ⭐️  this plugin on GitHub if you're finding it useful!
[04/11/2020, 10:13:46] [eWeLink] Note: 'Request & Response Logging' is not advised for long-term use.
    root : TTY=pts/0 ; PWD=/var/homebridge ; USER=root ; COMMAND=/bin/journalctl -o cat -n 500 -f -u homebridge
pam_unix(sudo:session): session opened for user root by (uid=0)
[04/11/2020, 10:13:51] [eWeLink] LAN message received.
{
  "deviceid": "**hidden**",
  "params": {
    "switch": "off",
    "sensorType": "AM2301",
    "humidity": 84,
    "temperature": 15,
    "updateSource": "LAN",
    "online": true
  }
}
[04/11/2020, 10:13:51] [eWeLink] [3D Drucker] LAN update received and will be refreshed.
[04/11/2020, 10:13:52] [eWeLink] LAN message received.
{
  "deviceid": "**hidden**",
  "params": {
    "switch": "off",
    "power": 0,
    "voltage": 232.21,
    "current": 0,
    "updateSource": "LAN",
    "online": true
  }
}
[04/11/2020, 10:13:52] [eWeLink] [Fenstersteckdosen] LAN update received and will be refreshed.
[04/11/2020, 10:13:52] [eWeLink] LAN message received.
{
  "deviceid": "**hidden**",
  "params": {
    "switch": "off",
    "updateSource": "LAN",
    "online": true
  }
}
[04/11/2020, 10:13:52] [eWeLink] [Küche] LAN update received and will be refreshed.
[04/11/2020, 10:13:52] [eWeLink] LAN message received.
{
  "deviceid": "**hidden**",
  "params": {
    "switch": "off",
    "updateSource": "LAN",
    "online": true
  }
}
[04/11/2020, 10:13:52] [eWeLink] [Sonoff-1] LAN update received and will be refreshed.
[04/11/2020, 10:13:52] [eWeLink] LAN message received.
{
  "deviceid": "**hidden**",
  "params": {
    "switch": "off",
    "updateSource": "LAN",
    "online": true
  }
}
[04/11/2020, 10:13:52] [eWeLink] [Alibert] LAN update received and will be refreshed.
[04/11/2020, 10:13:52] [eWeLink] LAN message received.
{
  "deviceid": "**hidden**",
  "params": {
    "switch": "off",
    "sensorType": "AM2301",
    "humidity": 60,
    "temperature": 17,
    "updateSource": "LAN",
    "online": true
  }
}
[04/11/2020, 10:13:52] [eWeLink] [Zirkulationspumpe] LAN update received and will be refreshed.
[04/11/2020, 10:13:52] [eWeLink] LAN message received.
{
  "deviceid": "**hidden**",
  "params": {
    "switch": "off",
    "updateSource": "LAN",
    "online": true
  }
}
[04/11/2020, 10:13:52] [eWeLink] [Spots Leinwand] LAN update received and will be refreshed.
[04/11/2020, 10:13:52] [eWeLink] LAN message received.
{
  "deviceid": "**hidden**",
  "params": {
    "switch": "off",
    "updateSource": "LAN",
    "online": true
  }
}
[04/11/2020, 10:13:52] [eWeLink] [Gang] LAN update received and will be refreshed.
```

Best 
Stephan

**Comment by bwp91** [comment]
Hi,

> but devices are not responding to on or off

What do you mean by this? All devices? Certain devices?

**Comment by SteeeveAT** [comment]
> Hi,
> 
> > but devices are not responding to on or off
> 
> What do you mean by this? All devices? Certain devices?

ALL Devices are not responding to any switch actions...

**Comment by bwp91** [comment]
Which switch device product models do you own?

**Comment by SteeeveAT** [comment]
* TH16
* S20
* POW R2
* SANTTON_3: (These guys): https://www.amazon.de/gp/product/B07ZNNRTD7/ref=ppx_yo_dt_b_search_asin_title?ie=UTF8&psc=1

**Comment by bwp91** [comment]
Do they work if you control them through the Homebridge UI rather than home app?

**Comment by SteeeveAT** [comment]
I just enabled this feature for this test. No, they sadly can't be controlled in the Accessorys tab. Same as in the home app - it looks like it is switched on, but it s'nt

**Comment by bwp91** [comment]
Would you be happy to share one of the devices to my ewelink account? Could we message on the [homebridge discord channel](https://discord.gg/7X36mdw)?

**Comment by SteeeveAT** [comment]
OK - while sorting the devices to rooms i found out, that most of the the ewelink devices are listed twice

**Comment by bwp91** [comment]
Original issue has been fixed in v3.11.0 and I believe we have fixed the further issues too, so going ahead and closing this issue.
thanks!
